lib/storage/remote.ts

```diff
@@ -44,18 +49,18 @@ export class StorageRemote extends StorageBase {
             baseUrl: this.baseUrl,
         });
 
-        this.get = promisify(req.get);
-        this.post = promisify(req.post);
+        this.get = promisify(req.get as any as (uri: string, options?: request.CoreOptions) => request.Response);
+        this.post = promisify(req.post as any as (uri: string, options?: request.CoreOptions) => request.Response);
```

Comment:
The type hack here is the second thing I'm not happy about. There are shenanigans at play with TS picking the last overload that shows up because it assumes that overload is the most general and all-encompassing overload.